tests/test_size_estimations.py

```diff
@@ -0,0 +1,103 @@
+import pyarrow as pa
+import pyarrow.parquet as papq
+import pytest
+
+from daft.daft import testing as native_testing_utils
+from daft.table.micropartition import MicroPartition
+
+pytest.skip(allow_module_level=True, reason="Skipping because these tests don't currently pass")
+
+
+def get_scantask_estimated_size(pq_path: str, size_on_disk: int, columns: list[str] | None = None) -> int:
+    """Retrieve the estimated size for reading a given Parquet file"""
+    return native_testing_utils.estimate_in_memory_size_bytes(str(pq_path), size_on_disk, columns=columns)
+
+
+def get_actual_size(pq_path: str, columns: list[str] | None = None) -> int:
+    # Force materializationm of the MicroPartition using `.slice`
+    mp = MicroPartition.read_parquet(str(pq_path), columns=columns)
+    return mp.slice(0, len(mp)).size_bytes()
+
+
+def assert_close(size_on_disk, estimated: int, actual: int, pct: float = 0.3):
+    assert (
+        abs(actual - estimated) / estimated < pct
+    ), f"Expected {size_on_disk / 1_000_000:.2f}MB file estimated vs actual to be within {pct * 100}%: {estimated / 1_000_000:.2f}MB vs {actual / 1000_000:.2f}MB ({((estimated - actual) / estimated) * 100:.2f}%)"
+
+
+@pytest.mark.parametrize("compression", ["snappy", None], ids=["snappy", "no_compression"])
+@pytest.mark.parametrize("use_dictionary", [True, False], ids=["dict", "no_dict"])
+def test_estimations_unique_strings(tmpdir, use_dictionary, compression):
+    pq_path = tmpdir / f"unique_strings.use_dictionary={use_dictionary}.compression={compression}.pq"
+    data = [f"{'a' * 100}{i}" for i in range(1000_000)]
+    tbl = pa.table({"foo": data})
+    papq.write_table(tbl, pq_path, use_dictionary=use_dictionary, compression=compression)
+
+    size_on_disk = pq_path.stat().size
+    assert_close(size_on_disk, get_scantask_estimated_size(pq_path, size_on_disk), get_actual_size(pq_path))
+
+
+@pytest.mark.parametrize("compression", ["snappy", None], ids=["snappy", "no_compression"])
+@pytest.mark.parametrize("use_dictionary", [True, False], ids=["dict", "no_dict"])
+def test_estimations_dup_strings(tmpdir, use_dictionary, compression):
+    pq_path = tmpdir / f"dup_strings.use_dictionary={use_dictionary}.compression={compression}.pq"
+    data = ["a" * 100 for _ in range(1000_000)]
+    tbl = pa.table({"foo": data})
+    papq.write_table(tbl, pq_path, use_dictionary=use_dictionary, compression=compression)
+
+    size_on_disk = pq_path.stat().size
+    assert_close(size_on_disk, get_scantask_estimated_size(pq_path, size_on_disk), get_actual_size(pq_path))
+
+
+@pytest.mark.parametrize("compression", ["snappy", None], ids=["snappy", "no_compression"])
+@pytest.mark.parametrize("use_dictionary", [True, False], ids=["dict", "no_dict"])
+def test_estimations_unique_ints(tmpdir, use_dictionary, compression):
+    pq_path = tmpdir / f"unique_ints.use_dictionary={use_dictionary}.compression={compression}.pq"
+
+    data = [i for i in range(1000_000)]
```

Comment:
This might not be very characteristic of parquet sizes with random integers.

Snappy compresses very well for scenarios like this since it'll encode the deltas, which in this case are all `1`s that will take up a single byte.

But doing something like `random.randint(0, 1000_000)` produces 50% larger file sizes, and `random.randint(0, 1000_000_000)` produces 100% larger file sizes.

Reply:
Can add more cases after as well